pallets/multisig/src/benchmarking.rs

```diff
@@ -0,0 +1,284 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(feature = "runtime-benchmarks")]
+use crate::*;
+use frame_benchmarking::benchmarks;
+use frame_system::RawOrigin;
+use pallet_identity::benchmarking::UserBuilder;
+
+pub type MultiSig<T> = crate::Module<T>;
+pub type Identity<T> = identity::Module<T>;
+
+fn generate_signers<T: Trait>(keys: &mut Vec<Signatory<T::AccountId>>, n: usize) {
+    for x in 0..n {
+        keys.push(Signatory::Account(
+            <UserBuilder<T>>::default().build("key", x as u32).account,
+        ));
+    }
+}
+
+fn get_last_auth_id<T: Trait>(signatory: &Signatory<T::AccountId>) -> u64 {
+    <identity::Authorizations<T>>::iter_prefix_values(signatory)
+        .into_iter()
+        .max_by_key(|x| x.auth_id)
+        .expect("there are no authorizations")
+        .auth_id
+}
+
+fn generate_multisig<T: Trait>(
+    alice: T::AccountId,
+    origin: T::Origin,
+    signers: Vec<Signatory<T::AccountId>>,
+) -> Result<T::AccountId, DispatchError> {
+    let num_of_signers = signers.len() as u64;
+    generate_multisig_with_signers::<T>(alice, origin, signers, num_of_signers)
+}
+
+fn generate_multisig_with_signers<T: Trait>(
+    alice: T::AccountId,
+    origin: T::Origin,
+    signers: Vec<Signatory<T::AccountId>>,
+    num_of_signers: u64,
+) -> Result<T::AccountId, DispatchError> {
+    let multisig = <MultiSig<T>>::get_next_multisig_address(alice.clone());
+    <MultiSig<T>>::create_multisig(origin.clone(), signers.clone(), num_of_signers)?;
+    for signer in signers {
+        let auth_id = get_last_auth_id::<T>(&signer);
+        <MultiSig<T>>::unsafe_accept_multisig_signer(signer, auth_id)?;
+    }
+    Ok(multisig)
+}
+
+benchmarks! {
+    _ {}
+
+    create_multisig {
+        // Number of signers
+        let i in 1 .. T::MaxSigners::get();
+
+        let caller = <UserBuilder<T>>::default().build_with_did("caller", 0);
+        let mut signers = vec![Signatory::from(caller.did())];
+        generate_signers::<T>(&mut signers, i as usize);
+    }: _(caller.origin, signers, i as u64)
+
+    create_or_approve_proposal_as_identity {
+        // Number of signers
+        let i in 1 .. T::MaxSigners::get();
```

Comment:
The complexity of this function doesn't depend on the number of signers so we can remove this variable and use a fixed number of signers (should be more than 1. 2 works)